statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,209 @@
+from __future__ import division
+
+__all__ = ["PoissonZeroInflated"]
+
+from statsmodels.discrete.discrete_model import CountModel
+
+class GenericZeroInflated(CountModel):
+    __doc__ = """
+    Generiz Zero Inflated model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+    def __init__(self, endog, exog, exog_infl=None, offset=None,
+                       exposure=None, missing='none', **kwargs):
+        super(GenericZeroInflated, self).__init__(endog, exog, offset=offset,
+                                                  exposure=exposure,
+                                                  missing=missing, **kwargs)
+
+        if exog_infl is None:
+            exog_infl = np.ones((endog.size, 1))
+        self.k_exog = exog.shape[1]
+        self.k_inflate = exog_infl.shape[1]
+
+    def loglike(self, params):
+        """
+        Loglikelihood of Generic Zero Inflated model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : float
+            The log-likelihood function of the model evaluated at `params`.
+            See notes.
+
+        Notes
+        --------
+        .. math:: \\ln L=\\sum_{i=1}^{n}\\left[-\\lambda_{i}+y_{i}x_{i}^{\\prime}\\beta-\\ln y_{i}!\\right]
+        """
+        return np.sum(self.loglikeobs(params))
+
+    def loglikeobs(self, params):
+        """
+        Loglikelihood for observations of Generic Zero Inflated model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : ndarray (nobs,)
+            The log likelihood for each observation of the model evaluated
+            at `params`. See Notes
+
+        Notes
+        --------
+        .. math:: \\ln L_{i}=\\left[-\\lambda_{i}+y_{i}x_{i}^{\\prime}\\beta-\\ln y_{i}!\\right]
+
+        for observations :math:`i=1,...,n`
+
+        """
+        params_infl = params[self.k_exog:]
+        params_main = params[:self.k_exog]
+
+        y = self.endog
+        w = self.model_infl.predict(params_infl)
+        llf_main = self.main_model.loglikeobs(params_main)
+
+        llf = np.zeros_like(y)
```

Comment:
it's better to add the dtype here, np.float64, otherwise it will break if endog is int